dndserver/handlers/lobby.py

```diff
@@ -74,3 +84,28 @@ def open_map_select(ctx, msg):
     req.ParseFromString(msg)
     res = SS2C_OPEN_LOBBY_MAP_RES()
     return res
+
+
+def auto_match(ctx, msg):
+    req = SC2S_AUTO_MATCH_REG_REQ()
+    req.ParseFromString(msg)
+    party = get_party(account_id=sessions[ctx.transport].account.id)
+    for user in party.players:
+        transport, session = get_user(account_id=user.account.id)
+        if req.mode == 1:
+            if len(party.players) > 1:
+                matchteam = SS2C_AUTO_MATCH_REG_TEAM_NOT(result=1, mode=req.mode)
+                header = make_header(matchteam)
+                transport.write(header + matchteam.SerializeToString())
+            matchmaking_users.append({"party": party, "difficulty": req.mode})
+        elif req.mode == 2:
+            try:
+                if len(party.players) > 1:
+                    res = SS2C_AUTO_MATCH_REG_TEAM_NOT(result=1, mode=req.mode)
+                    header = make_header(res)
+                    transport.write(header + res.SerializeToString())
+                matchmaking_users.remove({"party": party, "difficulty": req.mode})
+                return
+            except ValueError:
+                pass
```

Comment:
I would probably split this out as you are adding the party multiple times in the loop
```suggestion
    if req.mode == 1:
        matchmaking_users.append({"party": party, "difficulty": req.mode})
    elif req.mode == 2:
        try:
            matchmaking_users.remove({"party": party, "difficulty": req.mode})
        except ValueError:
            pass

    if len(party.players) > 1:
        for user in party.players:
            transport, session = get_user(account_id=user.account.id)
            if req.mode == 1:
                matchteam = SS2C_AUTO_MATCH_REG_TEAM_NOT(result=1, mode=req.mode)
                header = make_header(matchteam)
                transport.write(header + matchteam.SerializeToString())
            elif req.mode == 2:
                res = SS2C_AUTO_MATCH_REG_TEAM_NOT(result=1, mode=req.mode)
                header = make_header(res)
                transport.write(header + res.SerializeToString())
```